python/entities.py

```diff
@@ -14,6 +15,46 @@
 from jinja2 import utils
 
 
+class Property(object):
+    def __init__(self, remote_type, backref=None):
+        self.remote_type = remote_type
+        self.backref = backref
+
+
+class Meta(type):
+    def __new__(cls, name, bases, attrs):
+        scalar_properties = {}
+        recursive_properties = {}
+        attrs["_is_entity"] = True
+        for key, value in attrs.items():
+            if isinstance(value, Property):
+                if getattr(value.remote_type, '_is_entity', False):
+                    recursive_properties[key] = value.remote_type
+                else:
+                    scalar_properties[key] = value.remote_type
+        attrs["json_properties"] = scalar_properties
+        attrs["recursive_properties"] = recursive_properties
+        new_cls = super().__new__(cls, name, bases, attrs)
+        inner_init = new_cls.__init__  # type: ignore
+
+        def wrap_init(*args, **kwargs):
+            inner_init(*args, **kwargs)
+            object = args[0]
+            for prop in object.json_properties:
+                setattr(object, prop, None)
+            for prop in object.recursive_properties:
+                setattr(object, prop, None)
+
+        new_cls.__init__ = functools.update_wrapper(wrap_init, inner_init)  # type: ignore  # noqa
+        for key, value in attrs.items():
+            if isinstance(value, Property) and getattr(value.remote_type,
+                                                       '_is_entity', False) \
+               and value.backref:
+
+                value.remote_type.recursive_properties[value.backref] = new_cls
```

Comment:
it is not programming, it is art!!!

Reply:
all programming is art. some is just real ugly (and/or practical minded) art.